### PR TITLE
Switching to Pytorch's clang-formatter.

### PR DESCRIPTION
**Pull request by datumbox**
Fixing #2372 by using [option 2](https://github.com/pytorch/vision/issues/2372#issuecomment-713704789).

### CODE REVIEW

File: .circleci/config.yml.in [hunk 1]
```diff
@@ -127,9 +127,10 @@ jobs:
       - checkout
       - run:
           command: |
-            sudo apt-get update -y
-            sudo apt-get install -y clang-format
-            ./travis-scripts/run-clang-format/run-clang-format.py -r torchvision/csrc
+            curl https://oss-clang-format.s3.us-east-2.amazonaws.com/linux64/clang-format-linux64 -o clang-format
+            chmod +x clang-format
+            sudo mv clang-format /opt/clang-format
```
Comment (fmassa) [inline .circleci/config.yml.in:132]: nit: do we need to move this to `/opt/clang-format` if we can specify a `--clang-format-executable` in the script?

**Review by fmassa — APPROVED** [review]
Changes look good to me.

I don't know who maintains the `oss-clang-format` on PyTorch side though, so it would be good to let them know that we started using it.

cc @seemethere if you know who maintains it.

File: .circleci/config.yml.in (hunk 1 above)
Comment (datumbox) [reply .circleci/config.yml.in:132]: It's not really necessary. I did not want to leave a binary in the root of the project. Can make a change if you want.

Comment (fmassa) [reply .circleci/config.yml.in:132]: No worries, it was just for my information.

**Comment by mthrok** [comment]
@datumbox if you have time, could you port this to torchaudio and torchtext?

**Comment by datumbox** [comment]
@mthrok I can have a look. From what I see, the 2 projects don't have clang-format setup on CI. 

So basically the task looks like:
1. Create a circleci task
2. Copy the travis-scripts/run-clang-format files to audio/text
3. Run the script and fix any formatting issues
4. Submit a PR in each project

Does that sound OK?

**Comment by mthrok** [comment]
Both projects do have jobs for linting, and they should be identical.

The CI job is defined here;
https://github.com/pytorch/audio/blob/master/.circleci/config.yml.in#L586-L615

Here is the script used; (flake8 + clang_format)
https://github.com/pytorch/audio/blob/master/.circleci/unittest/linux/scripts/run_style_checks.sh

And the job looks like this.
https://app.circleci.com/pipelines/github/pytorch/audio/3651/workflows/c2b4fe73-871a-409a-b4a4-39f2442a37be/jobs/108109

So, changing either the config or the job script to run the same formatter should do.

**Pull request merged**